### PR TITLE
Return sequelize instance / dao after .sync and use promise.each instead...

### PR DESCRIPTION
**Pull request by BridgeAR**
... of .reduce

### CODE REVIEW

**Comment by BridgeAR** [comment]
This fixes #3530 #3531 with a cleaner solution.

**Comment by BridgeAR** [comment]
Are really all of these sync functions necessary or would it be possible to use a map in at least one or two cases? I'm uncertain in some cases and I guess @mickhansen @janmeier you should be able to tell right away if it's possible or not.

**Comment by mickhansen** [comment]
Both `sync` and `dropAll` deal with constraint so they absolutely have to be done in order.

File: lib/dialects/sqlite/query-interface.js [hunk 1]
```diff
@@ -32,9 +32,9 @@ module.exports = {
       var sql = self.QueryGenerator.removeColumnQuery(tableName, fields)
         , subQueries = sql.split(';').filter(function(q) { return q !== ''; });
 
-      return Utils.Promise.reduce(subQueries, function(total, subQuery) {
+      return Utils.Promise.resolve(subQueries).each(function(subQuery) {
         return self.sequelize.query(subQuery + ';', null, { raw: true});
```
Comment (mickhansen) [inline lib/dialects/sqlite/query-interface.js]: https://github.com/sequelize/sequelize/blob/master/lib/dialects/sqlite/query-generator.js#L324 order is pretty important here too.

**Comment by mickhansen** [comment]
The sqlite queries do some stuff where they drop and recreate tables, so serial order is important aswell.

**Comment by BridgeAR** [comment]
I guess this is ready to merge

File: lib/dialects/sqlite/query-interface.js [hunk 2]
```diff
@@ -32,9 +32,9 @@ module.exports = {
       var sql = self.QueryGenerator.removeColumnQuery(tableName, fields)
         , subQueries = sql.split(';').filter(function(q) { return q !== ''; });
 
-      return Utils.Promise.reduce(subQueries, function(total, subQuery) {
+      return Utils.Promise.each(subQueries, function(subQuery) {
```
Comment (mickhansen) [inline lib/dialects/sqlite/query-interface.js]: just use `Promise` rather than `Utils.Promise`

**Pull request merged**